lib/ABI/FunctionType/Conversion.cpp

```diff
@@ -136,7 +136,13 @@ TCC::tryConvertingRegisterArguments(const ArgumentRegisters &Registers) {
   abi::RegisterState::Map Map(model::ABI::getArchitecture(ABI.ABI()));
   for (const model::NamedTypedRegister &Reg : Registers)
     Map[Reg.Location()].IsUsedForPassingArguments = abi::RegisterState::Yes;
-  for (auto [Register, Pair] : ABI.enforceRegisterState(Map))
+  auto DeductionResults = ABI.tryDeducingRegisterState(Map);
+  if (!DeductionResults.has_value()) {
+    revng_log(Log,
+              "Register state deductions failed for the arguments of the "
+              "function. The ABI is probably miss-detected.");
+  }
+  for (auto [Register, Pair] : DeductionResults.value())
```

Comment:
Can you rephrase giving me the full context?

Reply:
If you remember, when designing the register state deductions for (initially) efa, we provided two modes of operation:
* hard enforcing - where ABI is always right and we just rewrite any contradicting information present in the function with what ABI provides without warning.
* soft deduction - where we are not 100% sure on the ABI and if there are any contradictions, we just bail.

Initially this code used the "hard" mode. But I decided to switch to the soft one. But since I might be missing something, I decided that it's better to have one more pair of eyes on this.

Reply:
So we lost the mode in which we forcefully suppress registers that are incompatible with the ABI?

Why? That was a useful feature and I don't remember we decide on this.

Reply:
The code for it still there, it's just that I don't feel like it's a good idea to do that by default. If anything going back to `enforceRegisterState` is an easy thing to do at any point.

Reply:
I'd like to have a `bool` somewhere to switch, then it can be unused for now but I don't want to lose the feature.

For what's the default, let's see. Whatever was before is fine.

Reply:
Switch sounds good :enterprise:

Reply:
Please take a look at the `ToCABIFunctionType` analysis now. I "upgraded" this `bool` into an `pipeline::Option` right away.

P. S. Feel free to close this if it looks good.